assets/js/modules/analytics-4/datastore/custom-dimensions.js

```diff
@@ -309,6 +378,53 @@ const baseSelectors = {
 			);
 		}
 	),
+
+	/**
+	 * Checks whether the provided custom dimension is being created.
+	 *
+	 * @since n.e.x.t
+	 *
+	 * @param {Object} state           Data store's state.
+	 * @param {string} customDimension Custom dimensions to check.
+	 * @return {boolean} True the provided custom dimension is being created, otherwise false.
+	 */
+	isCreatingCustomDimension( state, customDimension ) {
+		return !! state?.customDimensionsBeingCreated.includes(
+			customDimension
+		);
+	},
+
+	/**
+	 * Returns the error if encountered while creating the provided custom dimension.
+	 *
+	 * @since n.e.x.t
+	 *
+	 * @param {Object} state           Data store's state.
+	 * @param {string} customDimension Custom dimension to obtain creation error for.
+	 * @return {(Object|undefined)} Error object if exists, otherwise undefined.
+	 */
+	getCreateCustomDimensionError: createRegistrySelector(
+		( select ) => ( state, customDimension ) => {
+			const propertyID = select( MODULES_ANALYTICS_4 ).getPropertyID();
+
+			return select( MODULES_ANALYTICS_4 ).getErrorForAction(
+				'createCustomDimension',
+				[ propertyID, possibleCustomDimensions[ customDimension ] ]
+			);
+		}
+	),
+
+	/**
+	 * Determines whether the available custom dimensions are being synced.
+	 *
+	 * @since n.e.x.t
+	 *
+	 * @param {Object} state Data store's state.
+	 * @return {boolean} TRUE if the available custom dimensions are being synced, otherwise FALSE.
+	 */
+	isSyncingAvailableCustomDimensions( state ) {
+		return !! state?.isSyncingAvailableCustomDimensions;
+	},
```

Comment:
Hi @eugene-manuilov.

I actually had this same thought when @aaemnnosttv and I were discussing this approach internally. This was Evan's concern:

> I think `isSyncingAvailableCustomDimensions` may be necessary to avoid a race condition, but we can check during implementation. Basically, my concern is that if each component checks `isFetchingSyncAvailableCustomDimensions` before making the request, they'd all see that it is not being fetched and then invoke the fetch simultaneously. If we lock it with a simple state before the fetch which is set synchronously, I think that will be avoided.

Even though I wasn't able to replicate this race condition in my short span of tests, however, I think Evan's concern makes sense and a race condition might occur if we go down that path.

What do you think?

Reply:
How is `isSyncingAvailableCustomDimensions` different from `isFetchingSyncAvailableCustomDimensions`? These are completely similar selectors with exactly the same logic. If we worry that isFetching may cause race conditions, then isSyncing has the same problem. I think that if we truly want to be safe here, we should use debounce the sync action and call it in the create dimensions action.

cc @aaemnnosttv